core/validatorapi/validatorapi.go

```diff
@@ -357,6 +360,66 @@ func (c Component) SubmitBeaconBlock(ctx context.Context, block *spec.VersionedS
 	return nil
 }
 
+// SubmitVoluntaryExit receives the partially signed voluntary exit.
+func (c Component) SubmitVoluntaryExit(ctx context.Context, ve *eth2p0.SignedVoluntaryExit) error {
+	vs, err := c.Validators(ctx, "head", []eth2p0.ValidatorIndex{ve.Message.ValidatorIndex})
+	if err != nil {
+		return err
+	}
+	if len(vs) != 1 {
+		return errors.New("cannot find validator", z.U64("ValidatorIndex", uint64(ve.Message.ValidatorIndex)))
+	}
+
+	pubKeyBytes, err := vs[ve.Message.ValidatorIndex].PubKey(ctx)
```

Comment:
this will panic if the map contains a different index.

Reply:
can rename variable to `eth2Pubkey`

Reply:
Done both things